app/services/PaymentAPIService.scala

```diff
@@ -57,6 +73,26 @@ class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String) {
       .execute()
   }
 
+  def getPaypalApiError(body: String): Option[PaymentApiError] = {
+    val bodyJson = parse(body)
+    val error = bodyJson.toOption
+    val paypalApiError = error.flatMap(json => decodeErrorWrapper(json.hcursor).toOption)
+    paypalApiError.map(_.error)
+  }
+
+  def showAsSuccessful(response: WSResponse): Boolean = {
+    response match {
+      case response: WSResponse if response.status == 200 => true
+      case response: WSResponse if response.status == 400 => {
+        val error = getPaypalApiError(response.body)
```

Comment:
more idiomatic to do a `fold` on the `Option` (or `Either` if you use `decode` from circe to implement `getPaypalApiError`)